osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs

```diff
@@ -98,9 +101,13 @@ public override (Vector2 position, double time) GetSnappedPosition(Vector2 posit
             if (travelLength < DistanceBetweenTicks)
                 travelLength = DistanceBetweenTicks;
 
+            if (LimitedDistanceSnap.Value)
+                travelLength = SnapProvider.DurationToDistance(ReferenceObject, editorClock.CurrentTime - ReferenceObject.GetEndTime());
```

Comment:
This breaks with non-1.0x distance spacing.

https://github.com/ppy/osu/assets/20418176/fbd0beb5-a9bc-4082-aa2f-3f74f4887eb0

The snap doesn't snap to the brightest gridline.

Something like this appears to help:

```diff
diff --git a/osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs b/osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs
index 2eec833832..63886e38eb 100644
--- a/osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs
+++ b/osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs
@@ -101,12 +101,11 @@ public override (Vector2 position, double time) GetSnappedPosition(Vector2 posit
             if (travelLength < DistanceBetweenTicks)
                 travelLength = DistanceBetweenTicks;
 
-            if (LimitedDistanceSnap.Value)
-                travelLength = SnapProvider.DurationToDistance(ReferenceObject, editorClock.CurrentTime - ReferenceObject.GetEndTime());
-
-            // When interacting with the resolved snap provider, the distance spacing multiplier should first be removed
-            // to allow for snapping at a non-multiplied ratio.
-            float snappedDistance = SnapProvider.FindSnappedDistance(ReferenceObject, travelLength / distanceSpacingMultiplier);
+            float snappedDistance = LimitedDistanceSnap.Value
+                ? SnapProvider.DurationToDistance(ReferenceObject, editorClock.CurrentTime - ReferenceObject.GetEndTime())
+                // When interacting with the resolved snap provider, the distance spacing multiplier should first be removed
+                // to allow for snapping at a non-multiplied ratio.
+                : SnapProvider.FindSnappedDistance(ReferenceObject, travelLength / distanceSpacingMultiplier);
 
             double snappedTime = StartTime + SnapProvider.DistanceToDuration(ReferenceObject, snappedDistance);
 
```

but I'm not 100% confident so probably better you take a look first.

Reply:
The fix looks solid 👍